### ISSUE TITLE
Evaluate yet another review of JuMP Parameters 

### ISSUE THREAD

**Issue by jd-lara** [NREL-Sienna/PowerSimulations.jl#1126]
at JuMP dev the new syntax for parameters was shown and exists since JuMP 1.14. We should refactor some of the parameter handling in PSI that could benefit from better syntaxes. 

cc. @odow

**Comment by odow** [comment]
Yeah using `@variable(model, p in Parameter(1.0))` and `set_parameter_value`: https://jump.dev/JuMP.jl/stable/api/JuMP/#JuMP.set_parameter_value

should mean that we automatically add the `MOI.EqualTo` constraints if you use Gurobi and we automatically use parameters if you use POI or Ipopt.
